custom_components/battery_sim/__init__.py

```diff
@@ -203,188 +232,249 @@ def __init__(self, config, hass):
         }
 
         async_at_start(self._hass, self.async_source_tracking)
+
+        async_dispatcher_connect(
+            self._hass,
+            f"{self._name}-{MESSAGE_TYPE_GENERAL}",
+            self.async_reset_battery
+        )
+
         async_dispatcher_connect(
-            self._hass, f"{self._name}-BatteryResetMessage", self.async_reset_battery
+            self._hass,
+            f"{self._name}-{MESSAGE_TYPE_BATTERY_RESET_IMP}",
+            self.reset_sim_sensor,
         )
+
         async_dispatcher_connect(
             self._hass,
-            f"{self._name}-BatteryResetImportSim",
-            self.reset_import_sim_sensor,
+            f"{self._name}-{MESSAGE_TYPE_BATTERY_RESET_EXP}",
+            self.reset_sim_sensor,
         )
+
         async_dispatcher_connect(
             self._hass,
-            f"{self._name}-BatteryResetExportSim",
-            self.reset_export_sim_sensor,
+            f"{self._name}-{MESSAGE_TYPE_BATTERY_UPDATE}",
+            self.update_battery,
```

Comment:
This is causing an error. The battery update message is sent by the update_battery function when it has calculated all the variables to trigger the sensors to refresh not vice versa. Adding this listener is making it loop. Maybe we should rename it to "sensor update" or "battery has been updated". Anyway we don't want to listen for this in the __init__.py, only in the sensor.py so you can remove it from here. Thanks